### PR TITLE
Add reference line

### LINKED ISSUE #82: Add lines for number of days for doubling the cases
In this graph (https://aatishb.com/covidtrends/), can you add lines which indicate how many days it takes to double the cases? For eg, the line US is following indicates doubling every 5-6 days. A parallel line below it would indicate, say, 10 day doubling.
Regards,
Mandar

### PR DESCRIPTION
**Pull request by aatishb**
Closes #82.

I welcome feedback on this as this is making a fair number of changes. On the visual side, we are adding a reference line showing exponential growth with a user-selectable doubling time, as well as an annotation explaining what this line is. Also added the reference line to the URL load options, so you can add a custom reference line to the custom URLS.

reference line screenshot

### CODE REVIEW

**Comment by rpkoller** [comment]
I've gave the branch a quick spin. Overall I love the reference line enhancement. Functionwise I haven't found anything to complain. Works flawlessly. My suggestions are more in the presentational field. 

- I am uncertain if the extra month and day in graph is necessary. It is already in the controls underneath. the grid is reserved for the curves and the reference line in case it is activated. Aside that why in black, in case I would have expected it in red and with a heavier font weight.  ;) But never the less not really a fan of it there (but could live with it the longer i think about it) 

- Way less i could live with the explanation for the reference line in there. a legend within the graph isnt such a good idea imho :/ 

- I would make a two fold change or maybe threefold. First per default i would leave the show the x day doubling time deactivated. So people realize on activation what the purpose of that line is and how it looks. Second i would reword the label to something like show 2 day doubling time of covid19 population (or what ever works - i am not an english native speaker so it isn't that easy going for me to come up with something brief quickly ;)  - but i would just try to explain the part you had under the date combined with the  label text so the users understands by clicking this checkbox i get a line and this line symbolizes the curve representing the virus is doubling its population within 2 days. and third i would provide the more profound explanation for the moment in the header and at a later point on the possible future documentation page we discussed and mentioned briefly  in the a11y issue. 

- and i would perhaps try to colorize either the curves of the countries or instead the reference line. at the moment it is just black on black. and with many countries selected per default it is overwhelming 
Bildschirmfoto 2020-04-24 um 01 21 19

those are my two cents and the first thoughts coming to mind ;)

**Comment by aatishb** [comment]
@rpkoller thanks for this feedback. I hear you on the visually distracting annotation. Let me see what I can do about this. 

At the moment I feel a bit strongly about enabling this by default, because I think an exponential growth reference line is important to communicating the point of the graph (looking for deviations from exponential growth), and many people may not know to look for this.

The reason I opted for some on-screen annotation is that if people share screenshots of the graph, and the reference line is customizable, it's important that the image indicates what the line represents. Otherwise  if the line can change and it isn't labeled on the graph, I think we lose some consistency in visual communication.

**Comment by aatishb** [comment]
Apologies for the messy commit history. Attaching a screenshot of the current design.
Screen Shot 2020-04-24 at 7 45 54 PM

**Comment by rpkoller** [comment]
right before you published the screenshot i've switched to the reference line branch and took a look already. i like it much much much better now! 

the only suggestion i would voice is have you tried to change the color of the dotted doubling reference line? so you get a contrast to the uni color country curves?

**Comment by rpkoller** [comment]
the positioning of the label has also issues under certain circumstances 

Bildschirmfoto 2020-04-25 um 02 38 06

**Comment by aatishb** [comment]
Thanks for pointing that out, will look into it.

**Comment by rpkoller** [comment]
the label looks aligned now. and over all everything is still working as expected. dont see anything blocking left. i would merge it into `master`. afterwards you are still able to adjust and improve things further.

**Comment by aatishb** [comment]
@all-contributors add @rpkoller for review

**Comment by aatishb** [comment]
@all-contributor add @rpkoller for design

**Comment by aatishb** [comment]
Thanks @rpkoller for doing such a close review! Super appreciate it. :)

**Comment by rpkoller** [comment]
you are welcome... glad my comments helped a little bit...
